promptops/promptops/validation.py

```
@@ -14,9 +14,25 @@ class ComplexityLevel(str, Enum):
 class Message(BaseModel):
     role: Optional[str] = Field(None)
-    content: str = Field(...)
+    content: Optional[Union[str, List[Any]]] = Field(None)
+    tool_calls: Optional[List[ToolCall]] = Field(None)
+    tool_call_id: Optional[str] = Field(None)
+    name: Optional[str] = Field(None)
+
+    @model_validator(mode='after')
+    def check_content_or_tool_calls(self):
+        # Allow content to be missing/null if tool_calls is provided
+        if not self.content and not self.tool_calls and self.role != "tool" and self.role != "tool_result":
```

Ruff (0.15.15)

[warning] Consider merging multiple comparisons: `self.role not in {"tool", "tool_result"}`.

Merge multiple comparisons

(PLR1714)